### ISSUE TITLE
[TS] properly setup export of component props interfaces

### ISSUE THREAD

**Issue by Hyperkid123** [data-driven-forms/react-forms#544]
Component prop types are fragmented between multiple intersections. Only the component itself contains all necessary type definitions. Exported component props cannot be used to define the props object

```typescript
export interface SomeComponentProps { // this is missing additional intersection types
...
}

declare const SomeComponent: React.ComponentType<SomeComponentProps & UseFieldApiComponentConfig>

export default SomeComponent
```

The `SomeComponentProps` must include all unions to provide a full set of prop types for the interface.

**Comment by rvsia** [comment]
@Hyperkid123 This is resolved on 2.5.0 right?

**Comment by Hyperkid123** [comment]
Yes
